internal/testing/setup/setup.go

```diff
@@ -55,24 +59,75 @@ func NewAWSSession(t *testing.T, region string) (sess *session.Session, rt http.
 	}
 
 	client := &http.Client{Transport: r}
+	sess, err = awsSession(region, client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return sess, r, done
+}
 
+func awsSession(region string, client *http.Client) (*session.Session, error) {
 	// Provide fake creds if running in replay mode.
 	var creds *awscreds.Credentials
 	if !*Record {
 		creds = awscreds.NewStaticCredentials("FAKE_ID", "FAKE_SECRET", "FAKE_TOKEN")
 	}
-
-	sess, err = session.NewSession(&aws.Config{
+	return session.NewSession(&aws.Config{
 		HTTPClient:  client,
 		Region:      aws.String(region),
 		Credentials: creds,
 		MaxRetries:  aws.Int(0),
 	})
+}
+
+// NewAWSSession2 is like NewAWSSession, but it uses a diffrent record/replay proxy.
+func NewAWSSession2(t *testing.T, region string) (sess *session.Session, rt http.RoundTripper, done func()) {
+	httpreplay.DebugHeaders()
+	path := filepath.Join("testdata", t.Name()+".replay")
+	if *Record {
+		t.Logf("Recording into golden file %s", path)
+	} else {
+		t.Logf("Replaying from golden file %s", path)
+	}
+	if *Record {
+		if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
+			t.Fatal(err)
+		}
+		rec, err := httpreplay.NewRecorder(path, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		rec.RemoveQueryParams("X-Amz-Credential", "X-Amz-Signature")
+		rec.RemoveRequestHeaders("Authorization", "Duration", "X-Amz-Security-Token")
+		rec.ClearHeaders("X-Amz-Date")
+		rec.ClearHeaders("User-Agent") // AWS includes the Go version
```

Comment:
What does this new replayer use for matching? Does it include the `User-Agent` request header and that's why this needs to be cleared?

Reply:
My tool take the opposite approach of go-vcr: it matches all headers unless explicitly told not to. I feel that reduces the chance for false matches. So yes, if I didn't mention `User-Agent` here it would be matched, and that would fail.